baseplate/file_watcher.py

```diff
@@ -101,10 +88,6 @@ def __init__(self, path, parser, timeout=None, binary=False, encoding=None,
         self._mtime = 0
         self._data = _NOT_LOADED
         self._mode = "rb" if binary else "r"
-        # Since Python 2 does not support these kwargs, we store them as a dict
-        # that we `**` in the call to `open` in `get_data` so we do not have to
-        # call `open` in different ways depending on the Python version.  This
-        # can change if/when Python 2 support is dropped.
         self._open_options = {}
```

Comment:
Note, we can keep doing this or just store them as separate fields and pass them to open directly, i.e.:

```python
with open(self._path, self._mode, encoding=self._encoding, newline=self._newline) as f:
```

One thing that's kind of nice about this though is we don't have to care about what the default values are in python, so maybe it's worth maintaining this?

Reply:
Yeah I kinda like this, it also keeps all the open-related options in one place rather than cluttering up the instance.

Reply:
SGTM